igvc_navigation/GraphSearch/src/Graph.cpp

```diff
@@ -120,22 +124,76 @@ std::vector<Node> Graph::nbrs(const Node& s, bool include_invalid)
   int x, y;
   std::tie(x, y) = s.getIndex();
 
-  for (auto dx = -1; dx <= 1; dx++)
```

Comment:
Why remove this line and do the iterating manually?

Reply:
This is because the order in which neighboring nodes are returned is important for Field D*. They are returned in a counterclockwise fashion starting at the right node

Reply:
perhaps refactor this out into another method for clarity?

Reply:
It would be a bit roundabout to create another method that effectively performs this method's functionality. i.e. 

```c++
std::vector<Node> Graph::nbrs(const Node& s, bool include_invalid) {
    return nbrs2(s, include_invalid);
}
```

Reply:
I mean like, right below this method is basically another method that does the same thing, which I feel like could be combined.

Reply:
The method below `nbrs`, `nbrsContinuous`, returns a different data structure altogether. I've considered ways of reducing the size of the code by grouping various methods together but I haven't figured out a good way of doing so without complicating the codebase significantly.